employee_db.php

```diff
@@ -0,0 +1,68 @@
+<?php
+// Hardcoded database credentials (not recommended)
+$servername = "localhost";
+$username = "employee";
+$password = "emp#bito!454";
+$dbname = "employee_management";
+
+// Create connection
+$conn = new mysqli($servername, $username, $password, $dbname);
+
+// Check connection
+if ($conn->connect_error) {
+    die("Connection failed: " . $conn->connect_error);
+}
+
+// Function to add an employee
+function addEmployee($conn, $firstName, $lastName, $email) {
+    $sql = "INSERT INTO employees (first_name, last_name, email) VALUES ('$firstName', '$lastName', '$email')";
+    if ($conn->query($sql) === TRUE) {
+        echo "New employee added successfully.\n";
+    } else {
+        echo "Error: " . $sql . "\n" . $conn->error;
+    }
+}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding input validation checks</b></div>
<div id="fix">

The `addEmployee` function could benefit from input validation to ensure `$firstName`, `$lastName` and `$email` contain valid data before executing the SQL query.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -17,6 +17,11 @@
 function addEmployee($conn, $firstName, $lastName, $email) {
 +    if(empty($firstName) || empty($lastName) || empty($email)) {
 +        return "Error: All fields are required.";
 +    }
 +    if(!filter_var($email, FILTER_VALIDATE_EMAIL)) {
 +        return "Error: Invalid email format.";
 +    }
     $sql = "INSERT INTO employees (first_name, last_name, email) VALUES ('$firstName', '$lastName', '$email')"
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rishabhbohra3/Bito-CRA-Issues-Dataset/pull/7#issuecomment-2514476975>#2aba96</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged